var/spack/repos/builtin/packages/camp/package.py

```diff
@@ -55,6 +55,8 @@ class Camp(CMakePackage, CudaPackage, ROCmPackage):
 
     depends_on("blt", type="build")
 
+    patch("gcc-13.patch", when="@:2022.10.1 %gcc@13")
```

Comment:
OK, your suggestion is fair. I would rename the patch, though testing older versions of camp maybe a little hard: actually these headers are copied straightforward without a compile. It needs compile some other package using camp headers, e. g. umpire, to test if the patch works well.
Besides, It seems the patch is compatible in pattern at least with the 2020 "First numbered release"(https://github.com/LLNL/camp/releases/tag/v0.1.0), I think it would not be a big problem using @ :2022.10 .